pandas/tests/groupby/test_apply.py

```diff
@@ -961,3 +961,17 @@ def fn(x):
         name="col2",
     )
     tm.assert_series_equal(result, expected)
+
+
+def test_apply_function_with_indexing_return_column():
+    # GH: 7002
+    df = DataFrame(
+        {
+            "foo1": ["one", "two", "two", "three", "one", "two"],
+            "foo2": np.random.randn(6),
+        }
+    )
+    result = df.groupby("foo1", as_index=False).apply(lambda x: x.mean())
+    expected = df.groupby("foo1", as_index=False).mean()
```

Comment:
Can you explicitly construct the expected DataFrame here? Makes it easier to understand what's being tested and to constrain testing to a single implementation (groupby.apply). (Might be helpful to work with simple, deterministic test data for that.)

Reply:
I've fixed everything you suggested.